### ISSUE TITLE
<Plug>QfCprevious and similar commands doesn't work

### ISSUE THREAD

**Issue by nicoe** [romainl/vim-qf#9]
I don't know why but those map does not work:
```
nnoremap <Leader>h <Plug>QfCprevious
nnoremap <Leader>l <Plug>QfCnext
nnoremap <Leader>H <Plug>QfLprevious
nnoremap <Leader>L <Plug>QfLnext
```
While those do:
```
nnoremap <Leader>h :call qf#WrapCommand('up', 'c')<CR>
nnoremap <Leader>l :call qf#WrapCommand('down', 'c')<CR>
nnoremap <Leader>H :call qf#WrapCommand('up', 'l')<CR>
nnoremap <Leader>L :call qf#WrapCommand('down', 'l')<CR>
```
I though they were equivalent.

**Comment by nicoe** [comment]
I also noticed while debugging this that the test in WrapCommand `if exists('b:isLoc')` returns always false.

Maybe this is due to the fact I am using neovim instead of vim.

**Comment by ntnn** [comment]
`nnoremap` trasnlates to `normal mode no recursive mapping` - means
binds specified like that do not evaluate if what it is bound to is
bound to something else.

E.g.:
```vim
nnoremap $ ^
nnoremap x $x
nmap y $dl
```

The first line binds `$` (jump to eol) to `^` (jump to first non-empty char).
The second line changes `x` (delete char under cursor) to `$x` (jump to
eol and delete char under cursor). Here `$` keeps its meaning because
that mapping is non recursive.
The third line is recusrive, meaning that the `$` in the right hand side
is being evaluated for more mappings, so pressing `y` will move the
cursor to the first non-empty char in the line (because `$` is being
evaluated to the previous mapping in the first line) and then deletes
the char under the cursor (`dl` is equivalent to `x`, but would have
evaluated to `$x` because of the second line).

So, you have to use `nmap`, not `nnoremap`.

`b:isLoc` should be true if the windows is a location window, did you
test it after executing a location list command?

**Comment by romainl** [comment]
My bad, I used `nnoremap` all over the doc instead of `nmap`. Fixed as of commit 3389588.

**Comment by nicoe** [comment]
Thanks fo the info @ntnn I am always confused about the numerous `*map` commands.

About the `b:isLoc` issue, in fact I am using syntastic which automatically populates the location list once a file is save and there's some errors/warnings.

But when the `WrapCommand` is called the current window is obviously not a location window but the one where I edit the text (maybe I should open another issue instead of talking about that here).
